core/src/main/java/org/kohsuke/stapler/framework/io/LargeText.java

```diff
@@ -530,9 +532,14 @@ public void close() throws IOException {
             file.close();
         }
 
+        /**
+         * Like {@link RandomAccessFile#skipBytes} but with long instead of int.
+         */
         @Override
-        public void skip(long start) throws IOException {
-            file.seek(file.getFilePointer() + start);
+        public long skip(long start) throws IOException {
+            long to = Math.min(file.length(), file.getFilePointer() + start);
+            file.seek(to);
+            return to;
```

Comment:
This is supposed to be returning the number of bytes skipped, not the resulting offset, IIUC.

Reply:
Good spot, fixed in c964db5681e061f0915d611b804d47d2052fa59c. Thanks!

Side note: It did not affect the behavior. There is only one usage of skip() at the start of streaming. The file position is always 0 in this case.